### DISCUSSION TITLE
wslu 4.0.0

### DISCUSSION THREAD

**Discussion by car-meows** [wslutilities/wslu#240] [General]
##  :memo: Documentation

- Allow automatic manpage update
- Improve the contribution document

## :sparkles: Improvements

- wslu: `--debug` now usable; could work with `--verbose` properly now
- wslu: add Gentoo Linux, AlmaLinux, CBL Mariner, and Clear Linux support
- wslact: include new tool "memory reclaim"
- wslclip: brand new CLI (closes #221)
- wslfetch: add `-g, --generic`: this force the cli to use the generic WSL theme
- wslfetch: add `-t, --theme`: you can now provide your logo and color palette to use. also available as conf `WSLFETCH_THEME_PATH`
- wslfetch: add `-o, --options`: you can now customize what information to list out. also available as conf `WSLFETCH_INFO_SECTION`
- wslfetch: add conf `WSLFETCH_COLORBAR`: allows you to control color bar without passing param
- wslgsu: brand new CLI
- wslsys: add `-i, --ip`: get current WSL's IPv4 address. (closes #139 by #156 )
- wslsys: add `-T, --win-system-type`: get whether the WSL is running on Desktop or Server
- wslsys: add `-n, --name`: allow using the name to get the corresponding items
- wslsys: improvement on the speed to start up with a complete rewrite.
- wslusc: add `-s, --smart-icon`: add smart icon detection. Requires `wslpy`. can be set with conf `WSLUSC_SMART_ICON_DETECTION` (closes #176)
- wslusc: add `-d, --shortcut-debug`: allows you to view content of the generated shortcuts or any shortcuts
- wslusc: add `-N, --native`: allows you to generate shortcut using the native Wayland support. can be set with conf `WSLUSC_GUITYPE`
- wslvar: add configuration `WSLVAR_DEFAULT_VARTYPE`: this allows you to set default type of variable to get when not specifying option.
- wslview: add `-E, --engine`: you can now set the default method to start websites/folders/files. can be set with conf `WSLVIEW_DEFAULT_ENGINE` (closes #167, #197)

## :boom: Breaking Changes

- wslu: brand new configurations; default configuration located in /usr/share/wslu/conf
- wslu: individual version components are removed; they are hard to track
- wslu: ImageMagick is now a recommended dependency (closes #236 )
- wslact: `time-sync` is now `time-react`
- wslfetch: `-l, --line` and `-s, --splash` are removed with low usage

## :bug: Bug Fixes
- wslu: overall improvements on the code style to prevent problematic inputs
- wslview: redesigning the system to improve the detection (closes #181)

<hr /><em>This discussion was created from the release <a href='https://github.com/wslutilities/wslu/releases/tag/v4.0.0'>wslu 4.0.0</a>.</em>

**Comment by vbrozik** [comment]
It is great that the dependencies of the package were limited but I am not sure if it will help. In the version `3.2.4-0ubuntu3` the problematic dependencies seems to be `x11-utils` and `pulseaudio-utils`. `imagemagick` is only a suggested dependency in this version (not installed by default).

```
$ apt show wslu -a
Package: wslu
Version: 3.2.4-0ubuntu3
Priority: optional
Section: admin
Maintainer: Jinming Wu, Patrick <me@patrickwu.space>
Installed-Size: 216 kB
Provides: www-browser
Depends: desktop-file-utils, bc, psmisc
Recommends: pulseaudio-utils, x11-utils
Suggests: imagemagick
Download-Size: 81.3 kB
APT-Sources: https://ppa.launchpadcontent.net/wslutilities/wslu/ubuntu jammy/main amd64 Packages
Description: collection of utilities for Linux Subsystem for Windows (WSL).
 This is a collection of utilities for Linux Subsystem for Windows (WSL), such
 as for converting Linux paths to a Windows paths or creating Linux application
 shortcuts on the Windows Desktop.
 .
 Requires Windows 10 Creators Update and higher.
```

I noticed that the problem can be resolved by an `apt` option:
`sudo apt install --no-install-recommends wslu`
What about moving the recommended dependencies to the suggested category or mentioning this option in the installation instructions?

Please will the 4.x versions be released in the PPA soon?
https://launchpad.net/~wslutilities/+archive/ubuntu/wslu
